extensions/cornerstone-dicom-sr/src/utils/hydrateStructuredReport.js

```diff
@@ -101,6 +101,16 @@ export default function hydrateStructuredReport(
     metaData
   );
 
+  const onGenerateToolState =
+    customizationService.getModeCustomization('onGenerateToolState')?.value;
+
+  if (typeof onGenerateToolState === 'function') {
+    storedMeasurementByAnnotationType = onGenerateToolState({
+      storedMeasurementByAnnotationType,
+      displaySet,
+    });
+  }
```

Comment:
can you add a comment what is the typical usecase for this onGenerateToolState? I kind of prefer something related to `hydration`, since toolState is vaugly general